<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>5.- The Complex Microbiome</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<meta name="description" content="Turn your analyses into high quality documents, reports, presentations and dashboards with R Markdown. Use a productive notebook interface to weave together narrative text and code to produce elegantly formatted output. Use multiple languages including R, Python, and SQL. R Markdown supports a reproducible workflow for dozens of static and dynamic output formats including HTML, PDF, MS Word, Beamer, HTML5 slides, Tufte-style handouts, books, dashboards, shiny applications, scientific articles, websites, and more." />

<link rel="icon" type="image/png" href="images/favicon.png" />

<script type="text/javascript" src="js/redirect.js"></script>
<script type="text/javascript" src="js/rmarkdown.js"></script>
<script type="text/javascript" src="js/ga.js"></script>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
      .display.math{display: block; text-align: center; margin: 0.5rem auto;}
  </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="css/rmarkdown.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div id="rStudioHeader" class="alwaysShrunk">
  <div class="innards bandContent">
    <div>
      <a class="productName" href="index.html">Community Ecology and Biodiversity</a>
      <div class="Networks">    </div>
    </div>

    <div id="menu">
      <div id="menuToggler"></div>
      <div id="menuItems">
        <a class="menuItem" href="lesson-1.html">Get Started</a>
        
      </div>
    </div>
  </div>
</div>

<style type="text/css">
#TOC {
  margin-left: 35px;
  margin-top: 90px;
}
</style>

<script type="text/javascript">
$(".main-container").removeClass("main-container").removeClass("container-fluid").addClass("footerPushDown");
</script>


<div id="pageContent" class="standardPadding">
  <div class="articleBandContent">
<div class="lessonPage">
  <div class="lessonsNav">
    <a id="nav-lesson-1" href="lesson-1.html">Introduction</a>
    <a id="nav-lesson-2" href="lesson-2.html">1.- Network Structure: Construction and Metrics</a>
    <a id="nav-lesson-3" href="lesson-3.html">2.- The Assembly of Ecological Networks Across Spatial Scales</a>
    <a id="nav-lesson-4" href="lesson-4.html">3.- Effects of Global Change on Complex Communities</a>
    <a id="nav-lesson-5" href="lesson-5.html">4.- Ecological Applications</a>
    <a id="nav-lesson-6" href="lesson-6.html">5.- The Complex Microbiome</a>
   
  </div>
  <div class="lessonContent">

<div id="header">



<h1 class="title toc-ignore"><strong>5.- The Complex
Microbiome</strong></h1>

</div>


<p>Microbes are widespread across the globe, occupying a myriad of
ecological niches and found in environments as diverse as deep sea
vents, to soils and water, to inside the mammalian (including human)
gut. <strong>Microbes are not only responsible for a large portion of
many biogeochemical reactions that have enabled life as we know it,
including the global carbon cycle. They also have directly allowed the
existence of multicellular organisms</strong>* such as plants and
animals by creating intimate associations with them that allow them to
perform a suite of metabolic functions without which these multicellular
organisms wouldn’t be possible.</p>
<div id="free-living-bacterial-communities" class="section level2">
<h2>Free-living bacterial communities</h2>
<p>Microorganisms are ubiquitous across the world, having colonised
virtually all environments on Earth. Free-living bacterial communities
in the ocean and the soil play key roles in biogeochemical processes
including carbon and nutrient cycling. Given their central importance
for life on Earth, they have become a central focus of ecological
studies aimed at better understanding their composition, structure and
function. This would not only allow a better mechanistic understanding
of the assembly mechanisms behind their emergence, but also to be in a
better position to predict their potential responses to global
anthropogenic changes.</p>
<p><img src="lesson-images/tara-oceans.png"
style="float:right; padding-left:10px" width="500" /></p>
<p><strong>Tools available from classical community ecology are commonly
applied to microbial ecology</strong> datasets to quantify patterns of
diversity across scales. Measures of <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20%5Calpha"
alt="\alpha" title="\alpha" class="math inline" />, <img
style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20%5Cbeta"
alt="\beta" title="\beta" class="math inline" />, and <img
style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20%5Cgamma"
alt="\gamma" title="\gamma" class="math inline" /> diversity, as well as
community ordination techniques are routinely used to characterise
biodiversity and describe differences across communities found on
different locations. These patterns are also commonly related to
different environmental variables of interest using a suite of
statistical methods.</p>
<div style="text-align: right; margin-bottom:.3cm;">
<strong> Figure taken from Sunagawa et al. (2015) </strong>
</div>
<p>Measures of <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20%5Calpha"
alt="\alpha" title="\alpha" class="math inline" /> and <img
style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20%5Cgamma"
alt="\gamma" title="\gamma" class="math inline" /> diversity quantify
diversity (often in terms of species richnes, i.e. number of species) at
local and regional scales respectively. <img
style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20%5Cbeta"
alt="\beta" title="\beta" class="math inline" /> diversity on the other
hand is a measure of change or dissimilarity between local communities,
commonly used in community ecology to quantify the turnover or
replacement of species across local patches. The most commonly used
measure to quantify <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20%5Cbeta"
alt="\beta" title="\beta" class="math inline" /> diversity in microbial
ecology is the Bray-Curtis dissimilarity index, a statistic used to
quantify the compositional difference between communities based on the
species abundances (i.e. counts) at each community. For comparing two
communities <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20i" alt="i"
title="i" class="math inline" /> and <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20j" alt="j"
title="j" class="math inline" />, it is defined thus:</p>
<p><img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Cdisplaystyle%20BC_%7Bij%7D%20%3D%201%20-%20%5Cfrac%7B2C_%7Bij%7D%7D%7BS_i%20%2B%20S_j%7D"
alt="BC_{ij} = 1 - \frac{2C_{ij}}{S_i + S_j}"
title="BC_{ij} = 1 - \frac{2C_{ij}}{S_i + S_j}" class="math display" />
where <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20C_%7Bij%7D"
alt="C_{ij}" title="C_{ij}" class="math inline" /> is the sum of the
lesser values for only those species that are found in both sites, and
<img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20S_i" alt="S_i"
title="S_i" class="math inline" /> and <img
style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20S_j" alt="S_j"
title="S_j" class="math inline" /> are the total number of individuals
(or counts) found at sites <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20i" alt="i"
title="i" class="math inline" /> and <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20j" alt="j"
title="j" class="math inline" /> respectively.</p>
<p>The application of these tools to microbial communities in marine
environments has revealed that communities found in different oceans
across the globe are substantially different from one another
(challenging the hypotheses that the ocean is a well mixed environment),
and these differences seem to be driven by ocean temperatures.
Additionally, different environmental factors seem to play a role in
determining the taxonomic and gene functional composition of these
communities.</p>
</div>
<div id="host-associated-bacterial-communities" class="section level2">
<h2>Host-associated bacterial communities</h2>
<p>Life on Earth as we know it wouldn’t be possible without microbes.
Multicellular organisms depend fundamentally on the intimate
relationships they have developed with microbial communities for
survival and health. This has resulted in <a
href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1002533">our
bodies containing about 1.3 as many bacterial as human cells</a>, with
the corresponding metabolic function potential provided by these added
gene pool.</p>
<p>Recent advances in gene sequencing technologies have made possible to
characterise the sheer diversity of microbial communities that live in
association with their multicellular hosts. In the human microbiome
community analysis reveals that <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20%5Calpha"
alt="\alpha" title="\alpha" class="math inline" /> diversity varies
greatly amongst niches (i.e. body parts), whereas <img
style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20%5Cbeta"
alt="\beta" title="\beta" class="math inline" /> diversity across the
same sites in different individuals is more or less constant.
Differences in species richness about body parts are reinforced by the
differences found on community composition (using principal component
analysis) acorss sites. This suggest that the human microbiome is more
or less consisten across individuals but there are great variations
across different niches associated to different parts of the human body.
This makes intuitive sense as we would expect different ‘ecosystem
fucntions’ to be relevant in different parts of the body.</p>
<center>
<img src="lesson-images/human-microbiome.png" width="700" />
</center>
<p><strong>The organisation of the Human Microbiome</strong>.
<strong>(a)</strong> Alpha diversity within subjects by body habitat,
grouped by area. <strong>(b)</strong> Bray–Curtis beta diversity among
subjects by body habitat. <strong>(c)</strong> Principal coordinates
plot showing variation among samples demonstrates that primary
clustering is by body area, with the oral, gastrointestinal, skin and
urogenital habitats separate; the nares habitat bridges oral and skin
habitats. <strong>(d)</strong> Repeated samples from the same subject
(blue) are more similar than microbiomes from different subjects (red).
Technical replicates (grey) are in turn more similar; these patterns are
consistent for all body habitats and for both phylogenetic and metabolic
community composition. <strong><em>Figure and caption extracted from The
Human Microbiome Project Consortium (2012)</em></strong>.</p>
<p>A similar pattern to that observed in the human microbiome across
niches distributed across the human body are observed across species in
different genera of multicellular organisms. Bacterial communtiies found
across species are different across host species, with explicit links to
the functional capabilities of these communities (e.g. Lurgi et
al. 2019). When studying these associations, microbial ecologists look
not only at the differences between community composition (as done for
the marine and human microbiome communities above) but also at the
differences in relative abundances of different taxonomic groups across
different microbiomes.</p>
<center>
<img src="lesson-images/taxonomic-comp-sponges.png" width="700" />
</center>
<p><strong>Taxonomic profile of microbial communities in marine
sponges.</strong> Average phylum-level taxonomic profile of microbial
symbiont communities in 81 different sponge species, seawater and marine
sediments. Colour scale shows relative abundance in percentage within
each host species. <strong><em>Figure and caption extracted from Thomas
et al. (2016).</em></strong></p>
<p>This taxonomical compositional representation, jointly with knowledge
on the potential functional capabilities of different microbial phyla,
provides clues on the metabolic needs of different host species and
their differences. This can in turn be related to ecosystem level
thinking about different environmental niches occupied by different host
species.</p>
<div id="beyond-classical-community-analyses-the-network-approach"
class="section level3">
<h3>Beyond classical community analyses: The network approach</h3>
<p><img src="lesson-images/sponge-network.png"
style="float:left; padding-right:10px" width="300" /></p>
<p>Classical community analyses like those illustrated above on the
microbiome can be complemented with more network oriented approaches to
unveil further organisational patterns.</p>
<p>To do this, network representations of the microbiome are required.
One approach is to create a bipartite network of the network of hosts
and the microbes found within them. In this case the network does not
necessarily represent an ecological interaction but it nonetheless
represents the patterns of occurrence of microbial taxa across host
species.</p>
<p>Looking at the structure of this host-microbiome network can reveal
patterns of, for example, microbial persistence across host species.</p>
<p><img src="lesson-images/host-specificity-sponges.png"
style="float:right; padding-left:10px" width="400" /></p>
<p>In marine sponges, this analysis has revealed that <strong>the marine
sponge microbiome is made up of three kind of microbes. 1.- A set of
very specialised microbes that are found only a handful of hosts but
consistently within them (as measured by the fraction of samples on
their hosts in which they have been found). 2.- A group of ‘transient’
microbes that are found in an intermediate number of hosts but not very
frequently. And 3.- another set of very generalist microbes (i.e. found
in many different host species), which are again found relatively
consistent across them</strong>. This suggests that the composition of
the microbiome is made up of very specialised and very generalist
species with a set of microbes that come and go on different host
species, probably linked to exogenous stochastic factors.</p>
<p>Certain system-level network properties can provide insight into the
internal organisation of the microbiome can help us better understand
how microbial communities have self-organised across host species
(modularity); and how likely it is that microbial species found within
more specialised hosts are subsets of those found in generalist hosts,
thus hinting at a core microbioem that is conserved across species
(nestedness).</p>
</div>
<div id="modularity" class="section level3">
<h3>Modularity</h3>
<p>Modularity is a way of quantifying the extent to which a network is
organised into subcommunities within which their nodes are more
connected amongst themselves than to the rest of the network. It can be
quantified using the formula:</p>
<p><img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Cdisplaystyle%20M%20%3D%20%5Csum_%7Bs%3D1%7D%5E%7BN_M%7D%20%5Cleft%5B%20%5Cfrac%7Bl_s%7D%7BL%7D%20-%20%5Cleft%28%20%5Cfrac%7Bd_s%7D%7B2L%7D%20%5Cright%29%5E2%20%5Cright%5D"
alt="M = \sum_{s=1}^{N_M} \left[ \frac{l_s}{L} - \left( \frac{d_s}{2L} \right)^2 \right]"
title="M = \sum_{s=1}^{N_M} \left[ \frac{l_s}{L} - \left( \frac{d_s}{2L} \right)^2 \right]"
class="math display" /></p>
<p>where <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20N_M" alt="N_M"
title="N_M" class="math inline" /> is the number of modules, <img
style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20L" alt="L"
title="L" class="math inline" /> is the number of links in the network,
<img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20l_s" alt="l_s"
title="l_s" class="math inline" /> is the number of links between nodes
in module <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20s" alt="s"
title="s" class="math inline" />, and <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20d_s" alt="d_s"
title="d_s" class="math inline" /> is the sum of the degrees in module
<img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20s" alt="s"
title="s" class="math inline" />. Partitions of nodes into ‘modules’
that yield high scores for <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20M" alt="M"
title="M" class="math inline" /> ensure as many within-module links and
as few between-module links as possible. Finding partitions that improve
the value of <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20M" alt="M"
title="M" class="math inline" /> is a hard optimisation problem, and
different algorithms have been proposed in the literature to tackle
this. An implementation by Guimerà and Nunes Amaral (2005) using
simulated annealing yields robust results for this partitioning process.
They also define two futher metrics associated with the calculation of
modularity that can be useful to understand the role of each nodes in
the network. These are the within-module degree and the participation
coefficient. These measurements provide node-level information on the
connectivity profile at the module and network levels respectively. They
are defined thus:</p>
<p><img src="lesson-images/netcarto-roles.png"
style="float:right; padding-left:10px" width="300" /></p>
<p><img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Cdisplaystyle%20z_i%20%3D%20%5Cfrac%7Bk_i%20-%20%5Cbar%7Bk%7D_%7Bs_i%7D%7D%7B%5Csigma_%7Bk_%7Bs_i%7D%7D%7D"
alt="z_i = \frac{k_i - \bar{k}_{s_i}}{\sigma_{k_{s_i}}}"
title="z_i = \frac{k_i - \bar{k}_{s_i}}{\sigma_{k_{s_i}}}"
class="math display" /> where <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20z_i" alt="z_i"
title="z_i" class="math inline" /> is the within-module degree of node
<img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20i" alt="i"
title="i" class="math inline" />, <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20k_i" alt="k_i"
title="k_i" class="math inline" /> is the number of links to other nodes
in its module <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20s_i" alt="s_i"
title="s_i" class="math inline" />, <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20k_%7Bs_i%7D"
alt="k_{s_i}" title="k_{s_i}" class="math inline" /> is the average
degree of all nodes in <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20s_i" alt="s_i"
title="s_i" class="math inline" /> and <img
style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20%5Csigma_%7Bk_%7Bs_i%7D%7D"
alt="\sigma_{k_{s_i}}" title="\sigma_{k_{s_i}}" class="math inline" />
is the standard deviation of the degree of the nodes within <img
style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20s_i" alt="s_i"
title="s_i" class="math inline" />.</p>
<p>Participation coefficient, on the other hand, is defined as:</p>
<p><img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Cdisplaystyle%20P_i%20%3D%201%20-%20%5Csum_%7Bs%3D1%7D%5E%7BN_M%7D%20%5Cleft%28%20%5Cfrac%7Bk_%7Bis%7D%7D%7Bk_i%7D%20%5Cright%29%5E2"
alt="P_i = 1 - \sum_{s=1}^{N_M} \left( \frac{k_{is}}{k_i} \right)^2"
title="P_i = 1 - \sum_{s=1}^{N_M} \left( \frac{k_{is}}{k_i} \right)^2"
class="math display" /> where <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20k_%7Bis%7D"
alt="k_{is}" title="k_{is}" class="math inline" /> is the number of
links of node <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20i" alt="i"
title="i" class="math inline" /> to nodes in module <img
style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20s" alt="s"
title="s" class="math inline" /> and <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20k_i" alt="k_i"
title="k_i" class="math inline" /> is the total degree of node <img
style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20i" alt="i"
title="i" class="math inline" />.</p>
<p>The P-Z space can be classified into 7 roles that nodes can assume in
a network, as proposed by Guimerà and Nunes Amaral (2005), which allow
us to characterise nodes according to different profiles from
ultra-peripheral (nodes that are only connected within their modules) to
kinless hubs (nodes that are very connected to all modules in the
network).</p>
<p><strong>Modularity analysis performed over the global sponge
microbiome network has allowed to identify groups of host species that
harbour similar bacterial communities</strong>, along with the microbial
taxa that form these communities. Complementing this analysis with the
phylogenetic background of host species allowed us to unveil the
evolutionary drivers of these complex symbiotic associations.</p>
<center>
<img src="lesson-images/sponge-phylo-tree.png" width="500" />
</center>
<p><strong>Sponge phylogeny is associated with modularity in the global
sponge-core microbiome network.</strong> Sponge species belonging to the
same module identified over the sponge core–microbiome network display
the same colour. Stars, crosses and filled squares refer to sponge type
according to whether they are high- or low-microbial abundance species
(HMA and LMA respectively). NA = no information for host type is
available. <strong><em>Figure and caption extracted from Lurgi et
al. (2019).</em></strong></p>
<p><img src="lesson-images/p-vs-z-sponges.png"
style="float:right; padding-left:10px" width="300" /></p>
<p>Additionally, analysis of the structural roles of host species in the
host-microbiome network, revealed fundamental differences in the
organisation of High vs Low Microbial Abundance species. <strong>Sponge
species that harbour higher densities of microbes tend to be more
connected within their modules, whereas host species with low microbial
densities are well connected across the network</strong>. These suggests
that microbial communities in HMA sponges are more species- or
group-specific, while microbes found LMA sponges are more ubiquitous,
probably because of being more opportunistic and capable of dispersing
through the water column and colonising different species.</p>
</div>
<div id="nestedness" class="section level3">
<h3>Nestedness</h3>
<p>The observation that microbial communities in some host species are
more similar between groups of species, while other communities seem to
be composed of more ubiquitous microbes suggests a pattern of microbial
presence across hosts in which less microbially-rich hosts might harbour
bacterial communities that are subsets of more microbially diverse
hosts. One way of quantifying this potential pattern is by using a
measure of nestedness. Nestedness allows us to quantify the extent to
which the species composition (or composition interaction partners if
thinking of a plant animal interaction network) of less speciose
communities (or interacting partners of specialist species in a network)
are proper subsets of more diverse communities (or of the set of
interacting partners of generalist species if thinking about a bipartite
interaction network). This network property can be quantified using the
Nestedness metric based on Overlap and Decreasing Fill proposed by
Almeida-Neto et al. (2008):</p>
<p><img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Cdisplaystyle%20NODF%20%3D%20%5Cfrac%7B%5Csum%20N_%7Bpaired%7D%7D%7B%28%5Cfrac%7Bn%28n-1%29%7D%7B2%7D%29%2B%28%5Cfrac%7Bm%28m-1%29%7D%7B2%7D%29%7D"
alt="NODF = \frac{\sum N_{paired}}{(\frac{n(n-1)}{2})+(\frac{m(m-1)}{2})}"
title="NODF = \frac{\sum N_{paired}}{(\frac{n(n-1)}{2})+(\frac{m(m-1)}{2})}"
class="math display" /></p>
<p>where <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20N_%7Bpaired%7D"
alt="N_{paired}" title="N_{paired}" class="math inline" /> is a score
that quantifies the overlap between communities (or diets) of contiguous
rows and columns in a community matrix (or incidence matrix for
bipartite networks) and <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20n" alt="n"
title="n" class="math inline" /> and <img style="vertical-align:middle"
src="https://latex.codecogs.com/png.latex?%5Ctextstyle%20m" alt="m"
title="m" class="math inline" /> are the number of rows and columns in
the matrix respectively.</p>
<p>Using nestedness analysis on the microbiome can allow us to unveil
patterns in the way different taxonomic groups of species are organised
across different environments and host species. This approach has
revealed that across different microbiomes globally, species composition
in relatively species poor communities is a subset of the composition of
more diverse ones. Thus, microbiomes across environments seem to
increase in complexity starting from a core community of ubiquitously
found taxa, building up to new species that continue to aggregate in
into larger assemblages that contain the same species found in smaller
ones, possibly due to the availability of further ecological niches that
new taxa are able to colonise. Communities found in different
environments add new species depending on the availability ecological /
functional niches but conserving a set of core taxa, possibly generalist
taxa that are capble of surviving under different conditions.</p>
<center>
<img src="lesson-images/nestedness-microbial-communities.png"
width="700" />
</center>
<p><strong>Nestedness of community composition in microbiomes across
environments and multicellular host species.</strong>
<strong>(a)</strong> Presence–absence of phyla across samples, with
phyla (rows) sorted by prevalence and samples (columns) sorted by
richness. Shown is a subset of the EMP consisting of n=2,000
biologically independent samples with even representation across
environments and studies. With increasing sample richness (left to
right), phyla tended to be gained but not lost. <strong>(b)</strong> As
in <strong>a</strong> but separated into non-saline, saline, animal, and
plant environments. <strong>(c)</strong> Nestedness as a function of
taxonomic level, from phylum to tag sequence, across all samples and
within environment types. Also shown are median null model NODF scores
(± s.d.). NODF measures the average fraction of taxa from less diverse
communities that occur in more diverse communities. All environments at
all taxonomic levels were more nested than expected randomly, with
nestedness higher at higher taxonomic levels (for example, phyla).
<strong><em>Figure and caption from Thompson et
al. (2017).</em></strong></p>
</div>
</div>
<div id="modelling-the-microbiome" class="section level2">
<h2>Modelling the Microbiome</h2>
<p><img src="lesson-images/microbial-framework.jpeg"
style="float:right; padding-left:10px" width="300" /></p>
<p>Microbial communities within multicellular hosts are thought to form
complex networks of ecological interactions. They are amongst the most
complex ecological communities found in nature. It is thus an important
challenge to determine the strategies they might use to maintain
stability and persistence within their hosts. Modelling approaches like
the Lotka-Volterra framework that we explored in <a
href="lesson-4.html">section 3</a> can help us elucidate these
mechanisms.</p>
<p>Using a generalised Lotka-Volterra model for many species, Coyte et
al. (2015) explored different mechanisms by which hosts may control the
ecological interactions between microbial taxa within the microbiome.
These mechanisms, targeted at reducing positive feedbacks between
cooperative species (e.g. supplementary / targeted feeding by the host)
and decreasing the strength of interactions (e.g. via spatial
structuring) were shown to increase the stability of the microbiome.</p>
<p><img src="lesson-images/stability-microbial-networks.png"
style="float:left; padding-right:10px" width="300" /> <strong><em>Host
strategies to promote ecological stability.</em></strong>
<strong>(A)</strong> Spatial structure promotes ecological stability in
linear stability analysis (LAS) (left), permanence analysis (PA)
(middle), and an individual-based model (IbM) (right).
<strong>(B)</strong> Targeted feeding. Host-supplied nutrients that
weaken cooperative interactions promote ecological stability. We
observed an exception for which targeted feeding may destabilize more
competitive communities in the IbM (Pm = 0.25), but the effect is weak.
<strong><em>Figure and caption taken from Coyte et
al. (2015).</em></strong></p>
<p><br> <br> <br> <br></p>
</div>
<div id="references" class="section level2">
<h2>References</h2>
<p>Almeida-Neto M, Guimaraes P, Guimaraes PR, Loyola RD &amp; Ulrich W
(2008) A consistent metric for nestedness analysis in ecological
systems: reconciling concept and measurement.
<strong><em>Oikos</em></strong>, 117, 1227-1239.</p>
<p>Coyte KZ, Schluter J &amp; Foster KR (2015) The ecology of the
microbiome: Networks, competition, and stability.
<strong><em>Science</em></strong>, 350:6261, 663-666. <a
href="https://doi.org/10.1126/science.aad2602"
class="uri">https://doi.org/10.1126/science.aad2602</a></p>
<p>Guimerà R, Amaral LA. Cartography of complex networks: Modules and
universal roles. <strong><em>Journal of Statistical
Mechanics</em></strong>, P02001, 1-13. <a
href="https://doi.org/10.1088/1742-5468/2005/02/P02001"
class="uri">https://doi.org/10.1088/1742-5468/2005/02/P02001</a></p>
<p>Lurgi M, Thomas T, Wemheuer B, Webster NS &amp; Montoya JM (2019)
Modularity and predicted functions of the global sponge-microbiome
network. <strong><em>Nature Communications</em></strong>, 10, 992. <a
href="https://doi.org/10.1038/s41467-019-08925-4"
class="uri">https://doi.org/10.1038/s41467-019-08925-4</a></p>
<p>Sunagawa S et al. (2015) Structure and function of the global ocean
microbiome. <strong><em>Science</em></strong>, 348:6237. <a
href="https://doi.org/10.1126/science.1261359"
class="uri">https://doi.org/10.1126/science.1261359</a></p>
<p>The Human Microbiome Project Consortium. (2012) Structure, function
and diversity of the healthy human microbiome.
<strong><em>Nature</em></strong>, 486, 207–214. <a
href="https://doi.org/10.1038/nature11234"
class="uri">https://doi.org/10.1038/nature11234</a></p>
<p>Thomas T, Moitinho-Silva L, Lurgi M et al. (2016) Diversity,
structure and convergent evolution of the global sponge microbiome.
<strong><em>Nature Communications</em></strong> 7, 11870. <a
href="https://doi.org/10.1038/ncomms11870"
class="uri">https://doi.org/10.1038/ncomms11870</a></p>
<p>Thompson, L., Sanders, J., McDonald, D. et al. (2017) A communal
catalogue reveals Earth’s multiscale microbial diversity.
<strong><em>Nature</em></strong>, 551, 457–463. <a
href="https://doi.org/10.1038/nature24621"
class="uri">https://doi.org/10.1038/nature24621</a></p>
</div>
<div id="back-to-start" class="section level2">
<h2><a href="lesson-1.html" class="continue-link">Back to Start</a></h2>
</div>

  </div> <!-- lessonContent -->
</div> <!-- lessonPage -->


<script type="text/javascript">
  var lesson = window.location.href.match(/lesson-[0-9]+/g);
  if (lesson !== null) {
    lesson = 'nav-' + lesson[0];
    $('#'+lesson).addClass('current');
  }

  $('#show-answer').on("click", function() {
    $('#show-answer').addClass('showing');
    $('#model-answer').addClass('showing');
  })
</script>
  </div> <!-- articleBandContent -->
</div> <!-- pageContent -->

<div id="rStudioFooter" class="band full">
<div class="bandContent">
  <div id="copyright">© Copyright 2016 - 2020 RStudio, PBC</div>
  <div id="logos">
    <a href="https://twitter.com/rstudio" class="footerLogo twitter"></a>
    <a href="https://github.com/rstudio" class="footerLogo gitHub"></a>
    <a href="https://www.linkedin.com/company/rstudio-pbc" class="footerLogo linkedIn"></a>
    <a href="https://www.facebook.com/rstudioinc/" class="footerLogo facebook"></a>
  </div>
</div>
</div>

<!-- bizible -->
<script type="text/javascript" src="//cdn.bizible.com/scripts/bizible.js" async=""></script>

<!-- disqus -->
<div id="disqus_thread" class="standardPadding"></div>
<script type="text/javascript" src="js/disqus.js"></script>
<noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>

<script type="text/javascript" src="js/external-links.js"></script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->



</body>
</html>
